installer/environmentCheck/SystemValidator.php

```diff
@@ -0,0 +1,186 @@
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ *
+ */
+
+
+require __DIR__ . "/../../symfony/lib/vendor/autoload.php";
+class SystemValidator
+{
+    private $systemRequirements;
+
+    public function __construct()
+    {
+        $this->systemRequirements = sfYaml::load(file_get_contents(dirname(__FILE__) . '/system_requirements.yml'));
+
+    }
+
+    /**
+     * @return bool
+     */
+    public function isPhpCompatible()
+    {
+        $currentVersion = phpversion();
+        return $this->isWithinRange($currentVersion, $this->systemRequirements['phpversion']['excludeRange'], $this->systemRequirements['phpversion']['min'], $this->systemRequirements['phpversion']['max']);
+    }
+
+    /**
+     * @param $host
+     * @param $userName
+     * @param $password
+     * @return bool
+     */
+    public function isMySqlCompatible($host, $userName, $password)
+    {
+        $currentVersion = $this->getMySqlVersion($host, $userName, $password);
+
+        if (strpos($currentVersion, 'MariaDB') !== false) {
+            return $this->isWithinRange($this->getMariaDbVersion($currentVersion), $this->systemRequirements['mariadbversion']['excludeRange'], $this->systemRequirements['mariadbversion']['min'], $this->systemRequirements['mariadbversion']['max']);
+        }
+        return $this->isWithinRange($currentVersion, $this->systemRequirements['mysqlversion']['excludeRange'], $this->systemRequirements['mysqlversion']['min'], $this->systemRequirements['mysqlversion']['max']);
+    }
+
+    /**
+     * @return bool
+     */
+    public function isWebServerCompatible()
+    {
+        $webServer = $_SERVER['SERVER_SOFTWARE'];
+
+        foreach ($this->systemRequirements['webserver'] as $validServer) {
+            if (strpos($webServer, $validServer) !== false) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @return string
+     */
+    public function getPhpErrorMessage()
+    {
+        return $this->getErrorMessage('PHP', phpversion(), $this->systemRequirements['phpversion']['excludeRange'], $this->systemRequirements['phpversion']['min'], $this->systemRequirements['phpversion']['max']);
+    }
+
+    /**
+     * @param $host
+     * @param $userName
+     * @param $password
+     * @return string
+     */
+    public function getMysqlErrorMessage($host, $userName, $password)
+    {
+        $currentVersion = $this->getMySqlVersion($host, $userName, $password);
+        if (strpos($currentVersion, 'MariaDB') !== false) {
```

Comment:
the same condition repeated in another function as well. what if new private method like
isMariaDB()